### PR TITLE
MULE-17830: Fix properties not being propagated in basic auth

### PR DESCRIPTION
**Pull request by Pgarrett**

### CODE REVIEW

File: src/main/java/org/mule/extension/http/api/listener/HttpBasicAuthenticationFilter.java [hunk 1]
```diff
@@ -137,6 +143,13 @@ public void authenticate(AuthenticationHandler authenticationHandler)
     }
   }
 
+  private Map<String, Object> authenticationProperties(AuthenticationHandler authenticationHandler) {
+    return authenticationHandler.getAuthentication().isPresent()
```
Comment (afelisatti) [inline src/main/java/org/mule/extension/http/api/listener/HttpBasicAuthenticationFilter.java]: This seems much simpler `authenticationHandler.getAuthentication().map(Authentication::getProperties).orElse(emptyMap())`?

File: src/test/java/org/mule/test/http/api/HttpListenerAuthenticationHandlerTestCase.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    when(credentialsBuilder.withUsername(any())).thenReturn(credentialsBuilder);
+    DefaultMuleCredentials newCredentials = new DefaultMuleCredentials("user", "password".toCharArray());
+    when(credentialsBuilder.build()).thenReturn(newCredentials);
+    when(authenticationHandler.createCredentialsBuilder()).thenReturn(credentialsBuilder);
+    when(authenticationHandler.createAuthentication(newCredentials)).thenReturn(new DefaultMuleAuthentication(newCredentials));
+
+    securityProviders = new ArrayList<>();
+    attributes = mock(HttpRequestAttributes.class);
+    headers = new MultiMap<String, String>() {
+
```
Comment (afelisatti) [inline src/test/java/org/mule/test/http/api/HttpListenerAuthenticationHandlerTestCase.java]: Fix formatting

File: src/test/java/org/mule/test/http/api/HttpListenerAuthenticationHandlerTestCase.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class HttpListenerAuthenticationHandlerTestCase extends AbstractHttpAttributesTestCase {
```
Comment (afelisatti) [inline src/test/java/org/mule/test/http/api/HttpListenerAuthenticationHandlerTestCase.java]: Was there no way to test this with an integration test?

**Comment by afelisatti** [comment]
@Pgarrett please change the Jira reference to a MULE issue as this is an open source project requiring public tickets.

File: src/test/java/org/mule/test/http/api/HttpListenerAuthenticationHandlerTestCase.java (hunk 3 above)
Comment (Pgarrett) [reply src/test/java/org/mule/test/http/api/HttpListenerAuthenticationHandlerTestCase.java]: It was our first approach, but there was no easy way to set the authentication beforehand.

File: src/main/java/org/mule/extension/http/api/listener/HttpBasicAuthenticationFilter.java (hunk 1 above)
Comment (Pgarrett) [reply src/main/java/org/mule/extension/http/api/listener/HttpBasicAuthenticationFilter.java]: Done

File: src/test/java/org/mule/test/http/api/HttpListenerAuthenticationHandlerTestCase.java (hunk 2 above)
Comment (Pgarrett) [reply src/test/java/org/mule/test/http/api/HttpListenerAuthenticationHandlerTestCase.java]: The blank line is added by formatter

**Pull request merged**